heat/core/manipulations.py

```diff
@@ -59,9 +62,46 @@
     "unique",
     "vsplit",
     "vstack",
+    "to_sparse",
 ]
 
 
+def to_sparse(array: DNDarray) -> DCSR_matrix:
+    """
+    Convert the distributed array to a sparse DCSR_matrix representation.
+
+    Parameters
+    ----------
+    array : DNDarray
+        The distributed array to be converted to a sparse DCSR_matrix.
+
+    Returns
+    -------
+    DCSR_matrix
+        A sparse DCSR_matrix representation of the input DNDarray.
+
+    Notes
+    -----
+    This method allows for the conversion of a DNDarray into a sparse DCSR_matrix representation,
+    which is useful for handling large and sparse datasets efficiently.
+
+    Examples
+    --------
+    >>> dense_array = ht.array([[1, 0, 0], [0, 0, 2], [0, 3, 0]])
+    >>> sparse_matrix = dense_array.to_sparse()
+
+    """
+    array.balance_()
+    csr_matrices = [sparse_csr_matrix(tensor, is_split=array.split) for tensor in array.larray]
+    result = DCSR_matrix.from_csr_matrices(csr_matrices)
```

Comment:
Hi @samadpls!
You are in the right direction in that you have tried to preserve the split configuration using the `is_split` parameter. Kudos for remembering to balance the array first.

However, within a process, the whole data chunk must be treated as a single sparse tensor. So, this won't work,
```py
csr_matrices = [sparse_csr_matrix(tensor, is_split=array.split) for tensor in array.larray]
```
Also, `DCSR_matrix.from_csr_matrices` does not exist. If it is something that you intend to implement, I don't think there is a need for it.

After you balance the array, you can just use the existing factory method `factories.sparse_csr_matrix` with the correct split configuration. I think this would work for us.